tidy3d/plugins/mode/mode_solver.py

```diff
@@ -339,6 +353,339 @@ def data_raw(self) -> ModeSolverData:
+            # Rotate and apply translations
+            geometry = structure.geometry
+            geometry = (
+                geometry.translated(
+                    x=-translate_coords[0], y=-translate_coords[1], z=-translate_coords[2]
+                )
+                .rotated(theta, axis=self.bend_axis_3d)
+                .translated(x=translate_coords[0], y=translate_coords[1], z=translate_coords[2])
+            )
+
+            rotated_structures.append(structure.updated_copy(geometry=geometry))
+
+        return rotated_structures
+
+    @cached_property
+    def rotated_bend_center(self) -> List:
+        """Calculate the center of the rotated bend such that the modal plane is normal
+        to the azimuthal direction of the bend."""
+        rotated_bend_center = list(self.plane.center)
+        idx_rotate = 3 - self.normal_axis - self.bend_axis_3d
+        rotated_bend_center[idx_rotate] -= self.mode_spec.bend_radius
+
+        return rotated_bend_center
+
+    def _car_2_cyn(
+        self, mode_solver_data: ModeSolverData
+    ) -> Dict[Union[ScalarModeFieldCylindricalDataArray, ModeIndexDataArray]]:
+        """Convert Cartesian field data to cylindrical coordinates centered the
+        rotated bend center."""
+
+        # Extract coordinate values from one of the six field components
+        pts = [mode_solver_data.Ex[name].values.copy() for name in ["x", "y", "z"]]
+        f, mode_index = mode_solver_data.Ex.f, mode_solver_data.Ex.mode_index
+
+        idx_w, idx_uv = self.plane.pop_axis((0, 1, 2), axis=self.bend_axis_3d)
+        idx_u, idx_v = idx_uv
+
+        # offset coordinates by the rotated bend center in the bend plane
+        pts[idx_u] -= self.rotated_bend_center[idx_u]
+        pts[idx_v] -= self.rotated_bend_center[idx_v]
+
+        rho = np.sort(np.sqrt(pts[idx_u] ** 2 + pts[idx_v] ** 2))
+        theta = np.unique(np.arctan2(pts[idx_v], pts[idx_u]))
+        axial = pts[idx_w]
+
+        # Sanity check for theta uniqueness
+        if len(theta) > 1:
+            raise ValueError("Theta should have only one unique value.")
+
+        # Initialize output data arrays for cylindrical fields
+        field_data_cylindrical = {
+            field_name_cylindrical: np.zeros(
+                (len(rho), len(theta), len(axial), len(f), len(mode_index)), dtype=complex
+            )
+            for field_name_cylindrical in ("Er", "Etheta", "Eaxial", "Hr", "Htheta", "Haxial")
+        }
+
+        cos_theta, sin_theta = np.cos(theta), np.sin(theta)
+
+        axial_name, plane_names = self.plane.pop_axis(("x", "y", "z"), axis=self.bend_axis_3d)
+        cmp_1, cmp_2 = plane_names
+        sel_coords_func = {
+            cmp_1: lambda _rho: _rho * cos_theta + self.rotated_bend_center[idx_u],
+            cmp_2: lambda _rho: _rho * sin_theta + self.rotated_bend_center[idx_v],
+        }
+
+        # Transform fields from Cartesian to cylindrical coordinates
+        for i, _rho in enumerate(rho):
```

Comment:
I suspect this could be slow in some cases, but we may have no choice, or do you have any ideas about vectorizing?